File: kelly/6주차/min cost climbing stairs/solution02.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        # 전역 데이터
        n = len(cost)

        costs = [-1] * (n + 1)
        costs[0] = 0
        costs[1] = 0

        # 메인 로직
        for i in range(2, n + 1):
            costs[i] = min(costs[i - 1] + cost[i - 1], costs[i - 2] + cost[i - 2])

        return costs[n]

File: kelly/DP/min cost climbing stairs/Solution02.java
import java.util.*;

class Solution {
    public int minCostClimbingStairs(int[] cost) {
        Map<Integer, Integer> dp = new HashMap<>();
        dp.put(0, 0);
        dp.put(1, 0);

        for (int i = 2; i <= cost.length; i++) {
            dp.put(i, Math.min(dp.get(i - 1) + cost[i - 1], dp.get(i - 2) + cost[i - 2]));
        }

        return dp.get(cost.length);
    }
}